neural_compressor/compression/pruner/criteria.py

```diff
@@ -181,7 +182,10 @@ def __init__(self, modules, config):
         assert self.config.end_step > 0, "please set end_step > 0 for gradient based criterion"
         for key in modules.keys():
             p = modules[key].weight
-            self.scores[key] = torch.zeros(p.shape).to(p.device)
+            dtype = torch.float32
+            if self.low_memory_usage:
+                dtype = torch.bfloat16 if p.device.type == 'cpu' else torch.float16
```

Comment:
some cpu does not support bf16,  could you check the behavior,  will it fallback to fp32 or just throw exception